src/assert.ts

```diff
@@ -5,11 +5,15 @@ export function assertNever(value: never): never {
 }
 
 export function assertNodeKind<T extends FunctionlessNode>(
-  node: FunctionlessNode,
+  node: FunctionlessNode | undefined,
   kind: T["kind"]
 ): T {
-  if (node.kind !== kind) {
-    throw Error(`Expected node of type ${kind} and found ${node.kind}`);
+  if (node?.kind !== kind) {
+    throw Error(
```

Comment:
Should this be `new Error`?

Reply:
`Error()` and `new Error()` do the same thing.

Reply:
```ts
interface ErrorConstructor {
    new(message?: string): Error;
    (message?: string): Error;
    readonly prototype: Error;
}
```

Reply:
TIl